build/chip/chip_codegen.gni

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/build.gni")
+import("//build_overrides/chip.gni")
+import("//build_overrides/pigweed.gni")
+
+import("$dir_pw_build/python.gni")
+
+# Defines a target that runs code generation based on
+# scripts/codegen.py
+#
+# Arguments:
+#   idl
```

Comment:
Why not "dot_matter_file" or "idl_filename"?

Reply:
we generally use source/sources/output/outputs never source_filename or source_filenames or source_file.

I see no precedence for it to have a file/filename suffix and "dot_matter" or "matter" seemed vague/unclear. i also agree that `idl` may not be ideal if we evolve it, but I found no better way. Input?